src/java/org/apache/cassandra/io/sstable/format/big/BigTableWriter.java

```diff
@@ -439,14 +448,14 @@ protected BigTableWriter buildInternal(LifecycleNewTracker lifecycleNewTracker,
             {
                 mmappedRegionsCache = new MmappedRegionsCache();
                 rowIndexEntrySerializer = new RowIndexEntry.Serializer(descriptor.version, getSerializationHeader(), owner != null ? owner.getMetrics() : null);
-                dataWriter = DataComponent.buildWriter(descriptor,
-                                                       getTableMetadataRef().getLocal(),
-                                                       getIOOptions().writerOptions,
-                                                       getMetadataCollector(),
-                                                       lifecycleNewTracker.opType(),
-                                                       getIOOptions().flushCompression);
-                indexWriter = new IndexWriter(this);
-                partitionWriter = new BigFormatPartitionWriter(getSerializationHeader(), dataWriter, descriptor.version, rowIndexEntrySerializer.indexInfoSerializer());
+                dataWriter = LazyCloseableRef.create(() -> DataComponent.buildWriter(descriptor,
```

Comment:
Could this logic be in the `getDataWriter` method? (Renamed to `openDataWriter` for clarity.)